spacy/pipeline/entity_linker.py

```diff
@@ -235,14 +234,42 @@ def __init__(
         self.cfg: Dict[str, Any] = {"overwrite": overwrite}
         self.distance = CosineDistance(normalize=False)
         self.kb = generate_empty_kb(self.vocab, entity_vector_length)
-        self.scorer = scorer
         self.use_gold_ents = use_gold_ents
         self.candidates_batch_size = candidates_batch_size
         self.threshold = threshold
 
         if candidates_batch_size < 1:
             raise ValueError(Errors.E1044)
 
+        def _score_augmented(examples, **kwargs):
+            # Because of how spaCy works, we can't just score immediately, because Language.evaluate
+            # calls pipe() on the predicted docs, which won't have entities if there is no NER in the pipeline.
+            if not self.use_gold_ents:
+                return scorer(examples, **kwargs)
+            else:
+                examples = self._augment_examples(examples)
+                docs = self.pipe(
+                    (eg.predicted for eg in examples),
+                )
+                for eg, doc in zip(examples, docs):
+                    eg.predicted = doc
+                return scorer(examples, **kwargs)
+
+        self.scorer = _score_augmented
+
+    def _augment_examples(self, examples: Iterable[Example]) -> Iterable[Example]:
```

Comment:
I'm not very happy with the method name - this doesn't "augment" in the usual ML "data augmentation" sense, so this may be confusing. Maybe something like `_adopt_[predicted_]examples` or `_set_gold_ents`?

Reply:
True! I renamed it to `_ensure_ents` as the function doesn't set any ents if `use_gold_ents` is `False`.